### PR TITLE
Replace GSL with GSL-LITE submodule and fix up refs

### PR DESCRIPTION
**Pull request by yuslepukhin**
**Description**: GSL-LITE is a lightweight implementation of GSL that allows us to build source code with C++11 standard.

**Motivation and Context**
This change is a part of C++11 downgrade effort.
This also includes std::make_unique->onnxruntime::make_unique refactoring

### CODE REVIEW

File: include/onnxruntime/core/common/status.h [hunk 1]
```diff
@@ -86,14 +86,14 @@ class Status {
   Status(StatusCategory category, int code);
 
   Status(const Status& other)
-      : state_((other.state_ == nullptr) ? nullptr : std::make_unique<State>(*other.state_)) {}
+      : state_((other.state_ == nullptr) ? nullptr : new State(*other.state_)) {}
```
Comment (snnn) [inline include/onnxruntime/core/common/status.h]: The last time I did this,  Pranav rejected my change.

Comment (yuslepukhin) [reply include/onnxruntime/core/common/status.h]: I wonder why?

Comment (snnn) [reply include/onnxruntime/core/common/status.h]: https://github.com/microsoft/onnxruntime/pull/1730

Comment (pranavsharma) [reply include/onnxruntime/core/common/status.h]: @snnn The context was different. IIRC in your change, it was a factory function that was returning a raw ptr.

Comment (snnn) [reply include/onnxruntime/core/common/status.h]: Basically, if you are using operator 'new', you're outdated.

**Comment by yuslepukhin** [comment]
Closing for now.

File: cgmanifest.json [hunk 2]
```diff
@@ -31,8 +31,8 @@
          "component": {
```
Comment (yuslepukhin) [inline cgmanifest.json]: >"component": { [](start = 9, length = 14)

Will remove the component since we have a local copy.

Comment (snnn) [reply cgmanifest.json]: Please keep it. It is for helping us maintain and generating the https://github.com/microsoft/onnxruntime/blob/master/ThirdPartyNotices.txt

**Pull request merged**